### PR TITLE
Create room, roommate, housingType filter

### PR DESCRIPTION
**Pull request by Jiay0928**
Screen Shot 19

Screen Shot 17
Screen Shot 15

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt [hunk 1]
```diff
@@ -68,111 +70,124 @@ fun LocationFilter(
 		titleId = R.string.location,
 		closeAction = closeAction,
 		content = {
-			RangeSlider(
+			Column(
```
Comment (Jiay0928) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt:73]: Wrapping content with a column, everything else remains the same.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PriceFilter.kt [hunk 2]
```diff
@@ -46,40 +48,52 @@ fun PriceFilterForm(
 		titleId = R.string.price_range,
 		closeAction = closeAction,
 		content = {
-			Row(
+			Column(
```
Comment (Jiay0928) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PriceFilter.kt]: Wrapping content with a column, everything else remains the same.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt [hunk 3]
```diff
@@ -0,0 +1,225 @@
+package org.uwaterloo.subletr.pages.home.list.components
```
Comment (Jiay0928) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt:1]: Small code refactoring. Copy paste from the HomeListChildView.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt [hunk 4]
```diff
@@ -127,41 +135,52 @@ class HomeListChildViewModel @Inject constructor(
...
+		).map {
+			HomeListUiState.ListingItemsModel(
+				listings = it.first.listingParams.listingPagingParams.previousListingItemsModel.listings +
+					it.first.listingsResponse.listings,
+				likedListings = it.first.listingParams.listingPagingParams.previousListingItemsModel.likedListings +
+					it.first.listingsResponse.liked,
+				listingsImages = it.first.listingParams.listingPagingParams.previousListingItemsModel.listingsImages +
+					it.second,
+			)
+		}
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildViewModel.kt:152]: Gotta hate Kotlin sometimes lol
(No changes required, just ranting)

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt (hunk 3 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt:1]: Nice, looks good
(No changes required, just ranting)

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt:73]: Just out of curiosity, but why are we wrapping everything in a column?
I'm guessing that Compose is being weird and it fixes something #GottaLoveCompose

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/BasicFilterLayout.kt [hunk 5]
```diff
@@ -187,6 +195,29 @@ fun TextFieldWithErrorIndication(
...
+@Composable
+fun defaultFilterButton(
+	modifier: Modifier = Modifier,
+	isSelected: Boolean,
+	onClick: () -> Unit,
+	text: String,
+) {
+	SecondaryButton(
+		modifier = modifier
+			.defaultMinSize(minWidth = 1.dp, minHeight = 1.dp),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/BasicFilterLayout.kt]: Can you use `dimenResource` for this?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+					tint = secondaryTextColor,
+				)
+				Text(
+					text = listingSummary.residenceType.value,
+					style = listingDescriptionFont,
+				)
+			}
+			Row(
+				modifier = Modifier
+					.height(40.dp),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt:178]: Can't this just wrap content height?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PropertyTypeFilter.kt [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	closeAction: () -> Unit,
+) {
+	var housingPref by remember {
+		mutableStateOf(
+			currentHousingPref
+		)
+	}
+
+	BasicFilterLayout(
+		modifier = Modifier.height(280.dp),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PropertyTypeFilter.kt:33]: Do you need this or can we make it wrap content height?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoommateFilter.kt [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	updateGenderFilter: (Gender) -> Unit,
+	closeAction: () -> Unit,
+) {
+	var genderPref by remember {
+		mutableStateOf(
+			currentGenderPref
+		)
+	}
+	BasicFilterLayout(
+		modifier = Modifier.height(200.dp),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoommateFilter.kt:33]: Do you need this or can we make it wrap content height?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoomsFilter.kt [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+			currentRoomRange.bathroom
+		)
+	}
+	var ensuitebathroom by remember {
+		mutableStateOf(
+			currentRoomRange.ensuiteBathroom
+		)
+	}
+	BasicFilterLayout(
+		modifier = Modifier.height(500.dp),
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoomsFilter.kt:58]: Do you need this or can we make it wrap content height?

File: android/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -129,6 +129,14 @@
     <string name="km">km</string>
     <string name="show_sublets">Show sublets</string>
     <string name="price_range">Price Range</string>
+    <string name="bedrooms_for_sublet">Bedrooms for sublet</string>
+    <string name="bedrooms_in_property">Bedrooms in property</string>
+    <string name="bathrooms">Bathrooms</string>
+    <string name="any">Any</string>
+    <string name="five_plus">5+</string>
+    <string name="roommate_preference">Roommate Preference</string>
+    <string name="Apartment">Apartment</string>
```
Comment (leeminkyu16) [inline android/app/src/main/res/values/strings.xml]: Undercase `a` plz

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildView.kt [hunk 11]
```diff
@@ -192,117 +190,136 @@ fun HomeListChildView(
...
-								onClick = {
-									filterType.value = FilterType.LOCATION
-									coroutineScope.launch {
-										isBottomSheetOpen = true
+										}
+									},
+									contentDescription = stringResource(id = R.string.filter_menu),
+								)
+							}
+							item {
```
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildView.kt]: Why don't you do something like this here?
```kotlin
listOf(FilterType.LOCATION, ...).map { 
        item {
	        FilterButton(
		        filterName = stringResource(id = it.nameStringId),
		        onClick = {
			        filterType.value = it
			        coroutineScope.launch {
				        isBottomSheetOpen = true
        
			        }
		        },
	        )
        }
}
```
You can change the `FilterType` to also contain the `stringRes` for `filterName`

**Review by leeminkyu16 — COMMENTED** [review]
Just a few comments but overall looking good ^-^

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoommateFilter.kt (hunk 8 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoommateFilter.kt:33]: Wrap content height will make the bottom sheet to take up the entire screen.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoomsFilter.kt (hunk 9 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoomsFilter.kt:58]: No...

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PropertyTypeFilter.kt (hunk 7 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/PropertyTypeFilter.kt:33]: Wrap content height will make the bottom sheet to take up the entire screen...

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt (hunk 1 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/LocationFilter.kt:73]: Oh, I just made a bad decision of putting the column in the defaultFilter component. And I can't doing anything to change the configuration for that column, so I move it out.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt (hunk 6 above)
Comment (Jiay0928) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/ListingPost.kt:178]: Composable is adding random padding here. So, I just hardcode the height. (I rlly don't understand how size calculation works in composable).  (╯°□°）╯︵ ┻━┻

**Review by leeminkyu16 — APPROVED** [review]
Let's Gamble, Try Merging

**Review by andr-dev — APPROVED** [review]
Nice!

**Pull request merged**